debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/OracleChangeRecordValueConverter.java

```diff
@@ -233,14 +233,32 @@ private Object convertToLocalDateTime(Column column, Field fieldDefn, Object val
                 else {
                     dateTime = LocalDateTime.from(TIMESTAMP_FORMATTER.parse(dateText.trim()));
                 }
-                // todo: DBZ-137 this originally converted to ZoneId.systemDefault() but this should be GMT
-                // this is so that this behavior is compatible with the Xstreams implementation
-                return dateTime.atZone(ZoneId.of("GMT")).toInstant().toEpochMilli() * 1000; // timestamp(6) is converted as microTimestamp
+                return getDateTimeWithPrecision(column, dateTime);
+            }
+            else if (valueString.toLowerCase().startsWith("to_date")) {
+                dateText = valueString.substring("to_date".length() + 1, valueString.length() - 1);
+                String[] parts = dateText.split(",");
```

Comment:
Given this is on the hot path, it's worth extracting a pre-compiled `Pattern` which is kept as a static field.

Reply:
I assume you're referring specifically to the else-if check?